### PR TITLE
test: enable more extensive e2e testing for windows

### PR DESCRIPTION
**Pull request by jhockett**
#### Description of changes
- Convert current Windows e2e test allow list to a list of e2e tests to skip
  - This means by default, new e2e tests will run on Linux and Windows in CI
- Fixed a platform specific path issue in Lambda layer e2e tests
- Shortened the total text in some `.wait()` calls because our executors are injecting newlines to break up long lines of output

#### Description of how you validated changes
e2e tests passed: https://app.circleci.com/pipelines/github/aws-amplify/amplify-cli/13593/workflows/2d19fefa-7ac9-4d55-a291-b4653fbb5d6e

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are [changed or added](https://github.com/aws-amplify/amplify-cli/blob/dev/CONTRIBUTING.md#tests)
- [x] [Pull request labels](https://github.com/aws-amplify/amplify-cli/blob/dev/CONTRIBUTING.md#labels) are added

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: scripts/split-e2e-tests.ts [hunk 1]
```diff
@@ -14,83 +14,100 @@ const CONCURRENCY = 35;
...
-  'schema-model-b_pkg',
-  'schema-model-c_pkg',
-  'schema-model-d_pkg',
-  'schema-model-e_pkg',
-  'api_lambda_auth_pkg',
-  'node-function_pkg',
-  'schema-function-2_pkg',
-  'notifications_pkg',
-  'interactions_pkg',
+const WINDOWS_TEST_SKIP_LIST: string[] = [
```
Comment (awsluja) [inline scripts/split-e2e-tests.ts:17]: PR looks good; is there a way we can see the before & after of this? how many new tests were added & did we lost any tests unintentionally?

Can you also do 1 or 2 E2E run with retries disabled?
- this should help us identify any flaky windows tests before we merge them into the dev branch
i believe this is configured in the local_publish_helpers file

**Review by awsluja — APPROVED** [review]
LGTM

File: packages/amplify-util-uibuilder/src/commands/cloneComponentsFromEnv.ts [hunk 2]
```diff
@@ -11,7 +11,7 @@ export const run = async (context: $TSContext): Promise<void> => {
   const args = extractArgs(context);
   const sourceEnvName = args.sourceEnvName ? args.sourceEnvName : context.exeInfo.sourceEnvName;
   const newEnvName = args.newEnvName ? args.newEnvName : context.exeInfo.localEnvInfo.envName;
-  const appId = args.appId ?? context.exeInfo.teamProviderInfo[sourceEnvName].awscloudformation.AmplifyAppId;
+  const appId = args.appId ?? context.exeInfo?.teamProviderInfo?.[sourceEnvName]?.awscloudformation?.AmplifyAppId;
```
Comment (sobolk) [inline packages/amplify-util-uibuilder/src/commands/cloneComponentsFromEnv.ts:14]: is the code below tolerant to unknown appId ?
if not then we should validate.

**Review by sobolk — APPROVED** [review]

File: packages/amplify-util-uibuilder/src/commands/cloneComponentsFromEnv.ts (hunk 2 above)
Comment (jhockett) [reply packages/amplify-util-uibuilder/src/commands/cloneComponentsFromEnv.ts:14]: It is. There's a bunch of fallback logic for when appId is undefined.

Comment (jhockett) [reply packages/amplify-util-uibuilder/src/commands/cloneComponentsFromEnv.ts:14]: https://github.com/aws-amplify/amplify-cli/blob/dev/packages/amplify-util-uibuilder/src/clients/amplify-studio-client.ts#L80

File: scripts/split-e2e-tests.ts (hunk 1 above)
Comment (jhockett) [reply scripts/split-e2e-tests.ts:17]: Good catch, I had skipped the `analytics.test.ts` file by mistake on Windows. The diffs should now be distinct. I've also run this test suite a number of times so CCI should have some good data already on non-deterministic Windows tests.

**Review by awsluja — APPROVED** [review]

**Review by sobolk — APPROVED** [review]

**Pull request merged**